pages/index.js

```diff
@@ -1,69 +1,95 @@
-  return (
-    <div className={styles.container}>
-      <Head>
-        <title>Create Next App</title>
-        <meta name="description" content="Generated by create next app" />
-        <link rel="icon" href="/favicon.ico" />
-      </Head>
+  const [todos, setTodos] = useState([
+    { title: "test Todo", id: uuid(), done: true },
+  ]);
+
+  const [searchQuery, setSearchQuery] = useState("");
 
-      <main className={styles.main}>
-        <h1 className={styles.title}>
-          Welcome to <a href="https://nextjs.org">Next.js!</a>
-        </h1>
+  const handleDeleteTodo = (id) => {
+    setTodos(todos.filter((todo) => todo.id !== id));
+  };
 
-        <p className={styles.description}>
-          Get started by editing{' '}
-          <code className={styles.code}>pages/index.js</code>
-        </p>
+  const handleAddtodo = (event) => {
+    if (event.key === "Enter") {
+      setTodos([
+        ...todos,
+        { id: uuid(), title: event.target.value, done: false },
+      ]);
+      event.target.value = "";
+    }
+  };
 
-        <div className={styles.grid}>
-          <a href="https://nextjs.org/docs" className={styles.card}>
-            <h2>Documentation &rarr;</h2>
-            <p>Find in-depth information about Next.js features and API.</p>
-          </a>
+  // Abhacken von Todos
+  const toggleTodo = (id) => {
+    setTodos(
+      todos.map((todo) => {
+        if (todo.id === id) {
+          return { ...todo, done: !todo.done };
+        }
+        return todo;
+      })
+    );
+  };
 
-          <a href="https://nextjs.org/learn" className={styles.card}>
-            <h2>Learn &rarr;</h2>
-            <p>Learn about Next.js in an interactive course with quizzes!</p>
-          </a>
+  const activeTodos = todos.filter(({ done }) => !done);
+  const doneTodos = todos.filter(({ done }) => done);
+  const filterTodos = todos.filter((todo) =>
+    todo.title.toLowerCase().includes(searchQuery.toLowerCase())
+  );
+
+  // Suche von Todos
+  return (
+    <div className="main">
+      <div>
+        <input
+          onChange={(event) => setSearchQuery(event.target.value)}
+          type="text"
+          placeholder="search"
+        ></input>
+      </div>
 
-          <a
-            href="https://github.com/vercel/next.js/tree/canary/examples"
-            className={styles.card}
-          >
-            <h2>Examples &rarr;</h2>
-            <p>Discover and deploy boilerplate example Next.js projects.</p>
-          </a>
+      <div>
+        <h1>Meine Todo-App</h1>
+      </div>
 
-          <a
-            href="https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
-            className={styles.card}
-          >
-            <h2>Deploy &rarr;</h2>
-            <p>
-              Instantly deploy your Next.js site to a public URL with Vercel.
-            </p>
-          </a>
+      <div>
+        <div>
+          <input
+            onKeyDown={handleAddtodo}
```

Comment:
Please use `onChange` or `onInput`.
Please add value, to the "controlled input".

You can use a form to get ther native behavior instead of listening to the keydown event.